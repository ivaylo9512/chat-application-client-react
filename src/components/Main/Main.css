main{    
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: flex-end;
    flex: 1;
}
main p{
    margin: auto;
    font-weight: bold;
}