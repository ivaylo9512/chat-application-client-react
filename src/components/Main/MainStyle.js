import styled from "styled-components";

export const MainContainer = styled.div`    
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: flex-end;
    flex: 1;
`
export const P = styled.p`
    margin: auto;
    font-weight: bold;
    font-size: 1.3vw;
`