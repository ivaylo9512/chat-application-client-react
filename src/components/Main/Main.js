import { useState } from 'react'
import { Route, Redirect, Switch } from 'react-router-dom'
import ChatView from 'components/ChatView/ChatView'
import useEffectInitial from 'hooks/useEffectInitial';
import UserChatsView from 'components/UserChatsView/UserChatsView';
import UsersView from 'components/UsersView/UsersView'
import RequestsView from 'components/RequestsView/RequestsView'
import { MainContainer, P } from './MainStyle'

const Main = () => {
    const [webSocketClient, setWebSocketClient] = useState()

    return(
        <MainContainer>
            <Switch>
                <Route path='/searchUsers' render={() => <UsersView />}/>
                <Route path='/searchChat' render={() => <UserChatsView /> }/>
                <Route path='/chat' render={() => <ChatView />}/>
                <Route path='/requests' render={() => <RequestsView />}/>
                <Route path='/home' render={() => <P>Select a chat!</P>}/>
                <Redirect from='/' to='/home'/>
            </Switch>
        </MainContainer>
    )
}

export default Main