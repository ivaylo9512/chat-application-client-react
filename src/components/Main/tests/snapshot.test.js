import Main from 'components/Main/Main';
import { shallow } from 'enzyme';

describe('Main snapshot tests', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <Main />
        )
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})