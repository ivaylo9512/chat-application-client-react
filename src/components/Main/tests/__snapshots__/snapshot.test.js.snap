// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot tests should match snapshot 1`] = `
<styled.div>
  <Switch>
    <Route
      path="/searchUsers"
      render={[Function]}
    />
    <Route
      path="/searchChat"
      render={[Function]}
    />
    <Route
      path="/chat"
      render={[Function]}
    />
    <Route
      path="/home"
      render={[Function]}
    />
    <Redirect
      from="/"
      to="/home"
    />
  </Switch>
</styled.div>
`;
